pkg/prometheus/statefulset.go

```diff
@@ -599,7 +599,7 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 		thanosArgs := []string{
 			"sidecar",
 			fmt.Sprintf("--prometheus.url=http://%s:9090", c.LocalHost),
-			fmt.Sprintf("--tsdb.path=%s", storageDir),
+			fmt.Sprintf("--data-dir=%s", storageDir),
```

Comment:
This change is wrong. The flag only changed for the store:
> [breaking flag compatibility] Changed store gateway tsdb.path flag to data-dir

I've downloaded the 0.1.0 binary and run `thanos sidecar -h` where I can see `tsdb.path` as flag.